### PR TITLE
fix: PRT - Fix the bug that prevented existing addons to be enabled after spec  update

### PR DESCRIPTION
**Pull request by shleikes**
Co-authored with @Yaroms

### CODE REVIEW

File: protocol/chainlib/base_chain_parser.go [hunk 1]
```diff
@@ -242,11 +242,12 @@ func (bcp *BaseChainParser) Construct(spec spectypes.Spec, internalPaths map[str
...
 	allowedExtensions := map[string]struct{}{}
-	for _, apoCollection := range apiCollections {
-		for _, extension := range apoCollection.Extensions {
+	for _, apiCollection := range apiCollections {
+		for _, extension := range apiCollection.Extensions {
 			allowedExtensions[extension.Name] = struct{}{}
 		}
-		allowedAddons[apoCollection.CollectionData.AddOn] = false
+		// if addon was already existing (happens on spec update), use the existing policy, otherwise set it to false by default
+		allowedAddons[apiCollection.CollectionData.AddOn] = bcp.allowedAddons[apiCollection.CollectionData.AddOn]
```
Comment (ranlavanet) [inline protocol/chainlib/base_chain_parser.go:250]: can be bcp.allowedAddons can be nil. 

you need to iterate over bcp.allowedAddons and for any true value that also exists in allowedAddons set it to true.

**Review by ranlavanet — CHANGES_REQUESTED** [review]

File: protocol/chainlib/base_chain_parser.go [hunk 2]
```diff
@@ -242,11 +242,12 @@ func (bcp *BaseChainParser) Construct(spec spectypes.Spec, internalPaths map[str
 	bcp.verifications = verifications
 	allowedAddons := map[string]bool{}
 	allowedExtensions := map[string]struct{}{}
-	for _, apoCollection := range apiCollections {
-		for _, extension := range apoCollection.Extensions {
+	for _, apiCollection := range apiCollections {
+		for _, extension := range apiCollection.Extensions {
 			allowedExtensions[extension.Name] = struct{}{}
```
Comment (ranlavanet) [inline protocol/chainlib/base_chain_parser.go:247]: also you need a similar behavior to extensions.

call getConfiguredExtensions from the current bcp.extensionParser 
and set these extensions using 

func (ep *ExtensionParser) SetConfiguredExtensions(configuredExtensions map[ExtensionKey]*spectypes.Extension) {
	ep.configuredExtensions = configuredExtensions
}

**Review by ranlavanet — CHANGES_REQUESTED** [review]

File: protocol/chainlib/base_chain_parser.go (hunk 2 above)
Comment (shleikes) [reply protocol/chainlib/base_chain_parser.go:245]: No need, already happening. Made a small fix here:

**Review by ranlavanet — APPROVED** [review]

**Review by omerlavanet — APPROVED** [review]
this PR is 10/10 what a nice code

**Pull request merged**